src/UglyToad.PdfPig/Graphics/OperationContextHelper.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace UglyToad.PdfPig.Graphics
+{
+    using System;
+    using System.Linq;
+    using Content;
+    using Geometry;
+    using Logging;
+    using PdfPig.Core;
+
+    /// <summary>
+    /// Operation context helper methods.
+    /// </summary>
+    internal static class OperationContextHelper
+    {
+        /// <summary>
+        /// Get the initial transformation matrix.
+        /// </summary>
+        /// <param name="userSpaceUnit">User space unit.</param>
+        /// <param name="mediaBox">The Media box as define in the document, without any applied transform.</param>
+        /// <param name="cropBox">The Crop box as define in the document, without any applied transform.</param>
+        /// <param name="rotation">The page rotation.</param>
+        /// <param name="log"></param>
+        [System.Diagnostics.Contracts.Pure]
+        internal static TransformationMatrix GetInitialMatrix(UserSpaceUnit userSpaceUnit,
+            MediaBox mediaBox,
+            CropBox cropBox,
+            PageRotationDegrees rotation,
+            ILog log)
+        {
+            // Cater for scenario where the cropbox is larger than the mediabox.
+            // If there is no intersection (method returns null), fall back to the cropbox.
+            var viewBox = mediaBox.Bounds.Intersect(cropBox.Bounds) ?? cropBox.Bounds;
+
+            if (rotation.Value == 0
+                && viewBox.Left == 0
+                && viewBox.Bottom == 0
+                && userSpaceUnit.PointMultiples == 1)
+            {
+                return TransformationMatrix.Identity;
+            }
+
+            // Move points so that (0,0) is equal to the viewbox bottom left corner.
+            var t1 = TransformationMatrix.GetTranslationMatrix(-viewBox.Left, -viewBox.Bottom);
+
+            if (userSpaceUnit.PointMultiples != 1)
+            {
+                log.Warn("User space unit other than 1 is not implemented");
+            }
+
+            // After rotating around the origin, our points will have negative x/y coordinates.
+            // Fix this by translating them by a certain dx/dy after rotation based on the viewbox.
+            double dx, dy;
+            switch (rotation.Value)
```

Comment:
I'm not certain they exist but I think it would be possible to encounter multiples of 90 > 360, maybe switch on rotation.Value / 90 here (and then values 0, 1, 2, 3, 4)?